### DISCUSSION TITLE
Ubuntu 24: ERROR: Failed to open "My Pulse Output" (pulse); failed to connect: Connection refused

### DISCUSSION THREAD

**Discussion by maxbobse** [MusicPlayerDaemon/MPD#2126] [Q&A]
Hi,

I have installed mpd on Ubuntu 24.04.1 LTS and in my mpd.conf I have 

```
audio_output {
        type            "pulse"
        name            "My Pulse Output"
        server          "localhost"             # optional
##      sink            "remote_server_sink"    # optional
##      media_role      "media_role"            #optional
}
```
pulseausdio has been installed. If I now try to play a track I get the error message

```
ERROR: Failed to open "My Pulse Output" (pulse); failed to connect: Connection refused
```

Are there any recommendations how to use mpd with Ubuntu 24?

Max

**Comment by borine** [comment]
Remove (or comment out) the line:
```
        server          "localhost"             # optional
```
from your mpd.conf then restart mpd.

**Comment by urgor** [comment]
Then MPD shows
`exception: Failed to enable output "My Pulse Output" (pulse); pa_context_connect() has failed: Connection refused`
in logs.

**Reply by borine** [reply]
what are the outputs of
```console
ps -ef | grep pulseaudio
```
and
```console
ps -ef | grep mpd
```

**Reply by urgor** [reply]
$ps -ef | grep pulseaudio
```
urgor       4066    4039  0 13:38 ?        00:00:00 /usr/lib/x86_64-linux-gnu/xfce4/panel/wrapper-2.0 /usr/lib/x86_64-linux-gnu/xfce4/panel/plugins/libpulseaudio-plugin.so 12 14680082 pulseaudio PulseAudio Plugin Adjust the audio volume of the PulseAudio sound system
urgor       6362    6325  0 13:41 pts/1    00:00:00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox --exclude-dir=.venv --exclude-dir=venv pulseaudio
```


$ps -ef | grep mpd
```
mpd         6232       1  0 13:40 ?        00:00:00 /usr/bin/mpd --systemd
urgor       6424    6325  0 13:41 pts/1    00:00:00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox --exclude-dir=.venv --exclude-dir=venv mpd
```

**Reply by borine** [reply]
pulseaudio is not running on your system, perhaps you are using pipewire. What is the output of 
```console
ps -ef | grep pipewire
```

**Reply by urgor** [reply]
```
urgor       2410    2394  0 20:40 ?        00:00:00 /usr/bin/pipewire
urgor       2411    2394  0 20:40 ?        00:00:00 /usr/bin/pipewire -c filter-chain.conf
urgor       2413    2394  0 20:40 ?        00:00:00 /usr/bin/pipewire-pulse
urgor       4293    3077  0 20:43 pts/0    00:00:00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox --exclude-dir=.venv --exclude-dir=venv pipewire
```

As I understand it is kind of replacement for pulseaudio, and PA doesn't start automatically any more. What is the right approach:  enable PA daemon, or configure MPD to use Pipewire?

**Reply by borine** [reply]
I think you really need help from ubuntu. You can use pipewire with its own pulseaudio emulation, which is the usual way on desktop systems because there are not many pipewire control clients available, or you can have mpd output directly to pipewire.

This is not an mpd problem, please consult the ubuntu documentation and ask for help on ubuntu forums so that you can make the best decision for your own system.

**Reply by urgor** [reply]
Bornie, thank you for responses. You was right - not a mpd issue.

In the end i installed pulseaudio, removed pipewire using this [article from askubuntu](https://askubuntu.com/questions/1407885/how-to-uninstall-pipewire-and-go-back-to-pulseaudio), then added

`load-module module-native-protocol-tcp auth-ip-acl=127.0.0.1`

in /etc/pulse/default.pa  config, and 
```
audio_output {
        type            "pulse"
        name            "My Pulse Output"
        server          "127.0.0.1"
}
```
/etc/mpd.conf  config. And finally the best music player (mpd) works!

**Comment by thatsnek** [comment]
If anyone is reading this and using Ubuntu 24,
Following @urgor 's answer and this article: https://askubuntu.com/questions/1407885/how-to-uninstall-pipewire-and-go-back-to-pulseaudio

I installed the packages `sudo apt install pulseaudio pulseaudio-module-bluetooth pulseaudio-utils` and `sudo apt install pavucontrol`

Then masked the services `systemctl --user mask pipewire.service pipewire.socket wireplumber.service wireplumber@.service pipewire-pulse.service pipewire-pulse.socket`

Then rebooted my pc, then running `systemctl --user status pipewire` should say:

```
○ pipewire.service
     Loaded: masked (Reason: Unit pipewire.service is masked.)
     Active: inactive (dead)
```


once completed i followed @urgor 's steps by adding the load-module to /etc/pulse/default.pa then adding the pulse config then rebooted my PC and the error didint occur
